File: Lesson9/scrollviewapp/src/main/java/ru/mirea/gupalodr/scrollviewapp/MainActivity.java
package ru.mirea.gupalodr.scrollviewapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.math.BigInteger;

public class MainActivity extends AppCompatActivity {
// отобразить геометрическую прогрессию со знаменателем 2 до 100 элемента.
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        LinearLayout wrapper = findViewById(R.id.wrapper);
        BigInteger i_1 = BigInteger.valueOf(1);
        BigInteger q = BigInteger.valueOf(2);
        for (int i=1; i < 101; i++){
            View view = getLayoutInflater().inflate(R.layout.item, null, false);
            TextView text = (TextView)view.findViewById(R.id.textView);
            if (i==0) text.setText(String.format("Элемент %d = %d", i, 1));
            else text.setText(String.format("Элемент %d = %d", i, i_1.multiply(q)));
            wrapper.addView(view);
            i_1 = i_1.multiply(q);
        }

    }
}

File: Lesson3/favoritebook/src/main/java/ru/mirea/gupalodr/favoritebook/MainActivity.kt
package ru.mirea.gupalodr.favoritebook

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.activity.result.ActivityResult
import androidx.activity.result.ActivityResultCallback
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts

class MainActivity : AppCompatActivity() {
    companion object {
        const val KEY = "book_name"
        const val USER_MESSAGE = "MESSAGE"
    }

    private lateinit var  activityResultLauncher: ActivityResultLauncher<Intent>
    private lateinit var textViewUserBook: TextView
    private lateinit var button: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        textViewUserBook = findViewById(R.id.textViewBook)
        button = findViewById(R.id.button)

        val callback = ActivityResultCallback<ActivityResult> {
            if (it.resultCode == Activity.RESULT_OK) {
                val data = it.data
                val userBook = data?.getStringExtra(USER_MESSAGE)
                textViewUserBook.text = String.format("Название Вашей любимой книги: %s", userBook)

            }
        }
        activityResultLauncher = registerForActivityResult(
            ActivityResultContracts.StartActivityForResult(),
            callback
        )
        
        button.setOnClickListener {
            val intent = Intent(this, ShareActivity::class.java)
            intent.putExtra(KEY, "Метро")
            activityResultLauncher.launch(intent)
        }
    }
}